### PR TITLE
[Keyboard] Add splitkb.com's Aurora Sofle v2

### PR DESCRIPTION
**Pull request by harveysch**
Aurora Sofle v2

## Description

This pull request adds support for splitkb.com's latest product, the Aurora Sofle v2.

It is the fourth product in our Aurora range, which is expected to grow to five keyboard before the end of the year. They are redesigns of popular keyboards, reworked to comply with a set of standards we have come up with. By having all keyboards look and function the same, we are able to stock all parts for it efficiently and will be able to offer much better and consistent technical support, all while having an often better feature set as well.

The Aurora Sofle v2 is available for sale at [splitkb.com](https://splitkb.com/collections/keyboard-kits/products/aurora-sofle-v2/)

For more information about the Aurora series, please see our [October blog post](https://blog.splitkb.com/blog/october-2022)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/splitkb/aurora/sofle_v2/config.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// Not yet available in `info.json`
```
Comment (fauxpark) [inline keyboards/splitkb/aurora/sofle_v2/config.h:19]:
```suggestion
```

File: keyboards/splitkb/aurora/sofle_v2/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+// Not yet available in `keymap.json` format
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED
+#    define RGB_MATRIX_KEYPRESSES
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
+#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#    define ENABLE_RGB_MATRIX_RAINBOW_BEACON 
+#    define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#endif
```
Comment (fauxpark) [inline keyboards/splitkb/aurora/sofle_v2/keymaps/default/config.h:27]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
#define RGB_MATRIX_KEYPRESSES
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
```

File: keyboards/splitkb/aurora/sofle_v2/rev1/config.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+// Not yet available in `info.json`
+#ifdef BOOTMAGIC_ENABLE
+     // Top left key on left half
+#    define BOOTMAGIC_LITE_ROW 0
+#    define BOOTMAGIC_LITE_COLUMN 5
+     // Top right key on right half
+#    define BOOTMAGIC_LITE_ROW_RIGHT    5
+#    define BOOTMAGIC_LITE_COLUMN_RIGHT 5
+#endif
```
Comment (fauxpark) [inline keyboards/splitkb/aurora/sofle_v2/rev1/config.h:45]: These can go to info.json.

File: keyboards/splitkb/aurora/sofle_v2/rev1/rules.mk [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Although no rules are defined,
+# presence of this file is required for QMK to compile it.
```
Comment (fauxpark) [inline keyboards/splitkb/aurora/sofle_v2/rev1/rules.mk:17]:
```suggestion
```

File: keyboards/splitkb/aurora/sofle_v2/rev1/info.json [hunk 5]
```diff
@@ -0,0 +1,200 @@
+{
```
Comment (fauxpark) [inline keyboards/splitkb/aurora/sofle_v2/rev1/info.json:1]: Please run `qmk format-json` on all info.json files.

**Comment by thomasbaart** [comment]
Unrelated: How can we prevent these types of requested changes in the future? We try to stringently follow our previously reviewed and approved keyboard configurations, yet we're still faced with relatively surface-level comments each time.

We'll happily follow standards, but I'd prefer to know this upfront since we do take care when writing the configuration. Time is money, after all.

File: keyboards/splitkb/aurora/sofle_v2/keymaps/debug/keymap.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+bool oled_task_user(void) {
+    // A 128x32 OLED rotated 90 degrees is 5 characters wide and 16 characters tall
+    // This example string should fill that neatly
+    const char *text = PSTR("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ123456789!@#$%^&*()[]{}-=_+?");
+
+    if (is_keyboard_master()) {
+        oled_write_P(text, false);
+    } else {
+        oled_write_P(text, false);
+    }
```
Comment (drashna) [inline keyboards/splitkb/aurora/sofle_v2/keymaps/debug/keymap.c:72]:
```suggestion
    oled_write_P(PSTR("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ123456789!@#$%^&*()[]{}-=_+?"), false);
```

**Comment by fauxpark** [comment]
@thomasbaart the PR checklist should cover most of this, I think. I also have a revamped info.json reference coming soon (#21324) but it's got a small blocker at the moment.

**Comment by harveysch** [comment]
The licenses are the same as we have done for the previous keyboards, why do we need to do it different now?

File: keyboards/splitkb/aurora/sofle_v2/rev1/config.h (hunk 3 above)
Comment (harveysch) [reply keyboards/splitkb/aurora/sofle_v2/rev1/config.h]: RGB_MATRIX_LED_COUNT is not yet available in info.json as far as I can see?

**Comment by fauxpark** [comment]
License headers have never been needed for rules.mk.

**Comment by harveysch** [comment]
Anything that's still blocking this?

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by harveysch** [comment]
Anything left that's still blocking it?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**